aio_pika/robust_channel.py

```diff
@@ -121,7 +122,8 @@ def _on_channel_close(self, exc: Exception):
             if_unused=if_unused, nowait=nowait
         )
 
-        self._exchanges.pop(exchange_name, None)
+        self._exchanges[:] = filterfalse(lambda x: x.name == exchange_name,
```

Comment:
This change unneeded when `self._exchanges` is an instance of dict (defaultdict)

Reply:
I think it's less nice then just a list, but OK. updated the PR